### ISSUE TITLE
Desktop flickering throug the screencap

### ISSUE THREAD

**Issue by zufanka** [vkohaupt/vokoscreen#171]
[See the screencap](https://ufile.io/7ddfc) (link valid 30 days)

I am taking a screencap of the map, but my desktop image is flickering through the cap
![image](https://cloud.githubusercontent.com/assets/2411066/22126883/e205400e-de99-11e6-8376-c9b6463809e9.png)

I am taking an area, exporting to mp4 , but it does the same with any codec I use, if I capture window or area, I tried to change virtually any option it's still the same.

OS: Fedora 24

**Comment by tuxmuck** [comment]
I've never seen anything like this before -
I think this is a problem with your graphics card

**Comment by zufanka** [comment]
I checked now and every screencap software has the same problem.
Although I am quite sure it only happens since the system upgrade (Fedora
22 to 24), still on the same HW.
Any other tips maybe what it can be?

**Comment by debuglevel** [comment]
Hi zufanka,
tank you for your bug report.

do you see this flickering also on your physical screen or does it only happen on the recording?

Unfortunately, the upgrade from Fedora 22 to 24 is somewhat big. So it could be a bug in like everything.
I would guess this might be an issue in your graphics driver or the desktop environment. Maybe you could test with another ones, if you know how to so and if you really want to (to be honest, I have no idea how to change the graphics driver in modern Linux distributions. Probably messing around in some X11 config files).

**Comment by zufanka** [comment]
Hi Marc!

no, the flickering only happens in the recording, nothing out of the
ordinary happening on the physical screen.

Thank you, I will have a look into that :)

+31-652245912 | PGP 4D1819AD | @naberacka

**Comment by debuglevel** [comment]
Hi Adriana,
if it is only happening in the video file, it could also be a problem with ffmpeg (the underlying tool which does the recording).

There is a special version of vokoscreen which includes all the necessary dependencies (i.e. it brings its own version of ffmpeg): http://linuxecke.volkoh.de/vokoscreen/vokoscreen-download.html
You could give it a try.

**Comment by zufanka** [comment]
I did! But it's all the same

+31-652245912 | PGP 4D1819AD | @naberacka

**Comment by obilodeau** [comment]
I also have the same problem. Exact same behavior as the video.

* Arch Linux
* xf86-video-intel 2.99.917+747+g028c946d-1
* xorg-xserver 1.19.1-1
* Linux 4.9.5-1-ck-haswell
* QT 5.7.1

**Comment by vkohaupt** [comment]
@Adriane
what have your for a graphics card, Intel, Nvidia, AMD?

**Comment by vkohaupt** [comment]
@Olivier Bilodeau
what have you for a graphics card, Intel, Nvidia, AMD?

**Comment by obilodeau** [comment]
Intel:
```
        *-display
             description: VGA compatible controller
             product: 4th Gen Core Processor Integrated Graphics Controller
             vendor: Intel Corporation
             physical id: 2
             bus info: pci@0000:00:02.0
             version: 06
             width: 64 bits
             clock: 33MHz
             capabilities: msi pm vga_controller bus_master cap_list rom
             configuration: driver=i915 latency=0
             resources: irq:36 memory:f7400000-f77fffff memory:d0000000-dfffffff ioport:f000(size=64) memory:c0000-dffff
```

I noticed that using the Xorg intel TearFree option reduces significantly the amount of tearing:
```
$ cat /etc/X11/xorg.conf.d/20-intel.conf 
Section "Device"
	Identifier	"Intel Graphics"
	Driver		"intel"
	# attempting to solve screen recording glitches
	Option		"TearFree"	"true"
EndSection
```

**Comment by zufanka** [comment]
mine is Intel Corporation 3rd Gen Core processor Graphics Controller (rev
09)


+31-652245912 | PGP 4D1819AD | @naberacka

**Comment by vkohaupt** [comment]
Hallo Adriana and Oliver
perhaps can help the tip from my homepage.
http://linuxecke.volkoh.de/vokoscreen/vokoscreen-tips.html

**Comment by obilodeau** [comment]
There is no option like that in gnome. Not even in the gnome-tweak-tool.

**Comment by zufanka** [comment]
no, unfortunately I also can not find a way to do that in gnome

+31-652245912 | PGP 4D1819AD | @naberacka

On Fri, Feb 10, 2017 at 10:37 PM, Olivier Bilodeau <notifications@github.com

**Comment by albertostratio** [comment]
Hi @zufanka and @obilodeau ! Did you find a solution for this problem? I'm having the same issue.

**Comment by vkohaupt** [comment]
open a terminal and try this.

gsettings set org.gnome.desktop.interface enable-animations false

**Comment by Mrrvm** [comment]
I have the same issue, also using Arch with and Intel graphics card.

**Comment by zoliky** [comment]
I'm using Fedora 25 on a HP Probook 4540s notebook with Intel HD3000 graphics. I'm experiencing the same flickering problems, no matter what video capturing software I use. However, one of them, the GNOME "EasyScreenCast" extension records perfectly with no flickering. I'm still trying to figure out what does this GNOME extension do that other softwares don't.

**Comment by vkohaupt** [comment]
Hi all with Gnome and intel graphic card,

i deinstall the "xf86-video-intel" package from my opensuse 42.3 system.
Log out, log in or reboot the system, then make many screencasts and i 
see no more flickering.

Please test it and give a response

**Comment by obilodeau** [comment]
I switched from `intel` to `modesetting` with `AccelMethod` `glamor` a few months ago. I just tested a recording and it **does not flicker** indeed.

`/etc/X11/xorg.conf.d/20-graphics.conf`:
```
Section "Device"
	Identifier	"Intel Graphics"
	Driver		"modesetting"
	Option		"AccelMethod"    "glamor"
EndSection
```

**Comment by zoliky** [comment]
@obilodeau Fedora 26 uses modesetting by default now and there's no need to patch or remove anyting. However, on Fedora 25 I had to use your code and it worked well. Thanks for sharing!! There no "xf86-video-intel" on Fedora 25 and 26

**Comment by obilodeau** [comment]
@zoliky on Fedora the package is called: `xorg-x11-drv-intel` and starting with Fedora 26 [Xorg will use the modesetting driver instead of the intel one on recent GPUs](http://hansdegoede.livejournal.com/16976.html). You can also do the switch manually with the Xorg config I pasted above.

**Comment by aspiers** [comment]
I experienced flickering on openSUSE Leap 42.3 but eventually realised the problem was not with recording, but with playback via vlc.  When I started vlc with the `--avcodec-hw=vaapi` option the flicker vanished.  Then I added `avcodec-hw=vaapi` to the `[avcodec]` section of `~/.config/vlc/vlcrc` to make the option permanent.

**Comment by rakeshkumar125** [comment]
I have facing same type of problem. Please help us to resolve.
https://askubuntu.com/questions/1052981/i-have-facing-flicker-and-flashing-in-screen-recorded-video-with-ubuntu-16-4-del

Thanks in advance.

**Comment by rakeshkumar125** [comment]
Hello @obilodeau I have seen your comment on stackoverflow. 
The video link you provided does not work for me. Can you fix that please?
## I have shown you how recorded video output.

For Xorg, have you tried to add the TearFree option to your Xorg config as follows?
## Sorry Olivier but these files not avialble in my directory. 
[view directory](https://drive.google.com/file/d/1rmDARxfxgZgepnls-NdWU_jNEziSgktr/view)

**Comment by Vincent-CIRCL** [comment]
Same problem for me. 

> Linux user-pc 4.19.56-1-MANJARO #1 SMP PREEMPT Wed Jun 26 03:35:21 UTC 2019 x86_64 GNU/Linux

Removing "`xf86-video-intel`" indeed worked (`sudo pacman -R xf86-video-intel`)
Thanks a lot !

**Comment by tolistim** [comment]
I have uncovered a bit more information on this issue - it's related to the use of a compositor such as compviz.  I did some tests with 2 systems - one using Intel graphics and one using an Nvidia GTX 1070.  In both cases I have the proper vendor drivers installed.  The tearing occurs on both platforms when using KDE or Cinnamon.  However, by moving to Xfce or MATE (which don't use a compositor), the tearing is alleviated.

I suspect that in  the above situations where removing the Intel driver solved the issue was actually because the OSS driver doesn't support a compositor forcing the DM into software render mode.

I tested this on Mint 19.1, CentOS 7.6, SuSE LEAP Desktop, and Fedora 30.

I also agree that the problem is within ffmpeg (3.x and 4.x) and not Vokoscreen since I was able to see the same tearing when just using ffmpeg from a terminal.
